src/shared/modules/user/user.controller.ts

```diff
@@ -0,0 +1,97 @@
+import { inject, injectable } from 'inversify';
+import { Response } from 'express';
+import { StatusCodes } from 'http-status-codes';
+
+import { BaseController, HttpError, HttpMethod } from '../../libs/rest/index.js';
+import { Logger } from '../../libs/logger/index.js';
+import { Component } from '../../types/index.js';
+import { UserService } from './user-service.interface.js';
+import { fillDTO } from '../../helpers/index.js';
+import { UserRdo } from './rdo/user.rdo.js';
+import { RestSchema } from '../../libs/config/index.js';
+import { Config } from '../../libs/config/index.js';
+import { CreateUserRequest } from './create-user-request.type.js';
+import { LoginUserRequest } from './login-user-request.type.js';
+
+
+@injectable()
+export class UserController extends BaseController {
+  constructor(
+    @inject(Component.Logger) protected readonly logger: Logger,
+    @inject(Component.UserService) private readonly userService: UserService,
+    @inject(Component.Config) private readonly config: Config<RestSchema>,
+  ) {
+    super(logger);
+
+    this.logger.info('Register routes for UserController…');
+
+    this.addRoute({ path: '/register', method: HttpMethod.Post, handler: this.create });
+    this.addRoute({ path: '/login', method: HttpMethod.Post, handler: this.login });
+    this.addRoute({ path: '/login', method: HttpMethod.Get, handler: this.check });
+    this.addRoute({ path: '/logout', method: HttpMethod.Post, handler: this.logout });
+    this.addRoute({ path: '/:id/avatar', method: HttpMethod.Post, handler: this.uploadAvatar });
+  }
+
+  public async login(
+    { body }: LoginUserRequest,
+    _res: Response,
+  ): Promise<void> {
+    const existsUser = await this.userService.findByEmail(body.email);
+
+    if (! existsUser) {
+      throw new HttpError(
+        StatusCodes.UNAUTHORIZED,
+        `User with email ${body.email} not found.`,
+        'UserController',
+      );
+    }
+
+    throw new HttpError(
+      StatusCodes.NOT_IMPLEMENTED,
+      'Not implemented',
+      'UserController',
+    );
+  }
+
+  public async create(
+    { body }: CreateUserRequest,
+    res: Response,
+  ): Promise<void> {
+    const existsUser = await this.userService.findByEmail(body.email);
+
+    if (existsUser) {
+      throw new HttpError(
+        StatusCodes.CONFLICT,
+        `User with email «${body.email}» exists.`,
+        'UserController'
+      );
+    }
+
+    const result = await this.userService.create(body, this.config.get('SALT'));
+    this.created(res, fillDTO(UserRdo, result));
+  }
+
+  public async check(): Promise<void> {
+    throw new HttpError(
+      StatusCodes.NOT_IMPLEMENTED,
+      'Not implemented',
+      'UserController',
+    );
+  }
+
+  public async logout(): Promise<void> {
+    throw new HttpError(
+      StatusCodes.NOT_IMPLEMENTED,
+      'Not implemented',
+      'UserController',
```

Comment:
Если хочешь передать названия класса, лучше этого делать используя свойство name у класса. Таким образом, если менять названия класса или прямо удалить его, то бросится ошибка.
```suggestion
      UserController.name,
```